README.md

```diff
@@ -1,81 +1,164 @@
 
-## How to build
-To compile open_agb_firm, the following needs to be installed:
-* [devkitARM](https://sourceforge.net/projects/devkitpro/)
-* [Corelink DMA-330 assembler](https://github.com/profi200/dma330as)
-* [CTR firm builder](https://github.com/derrekr/ctr_firm_builder)
-
-Additionally, 7-Zip (or on Linux, p7zip) needs to be installed to make release builds. Also make sure the dma330as and CTR firm builder binaries are in the $PATH environment variable and accessible to the Makefile. Build open_agb_firm as a debug build via `make` or as a release build via `make release`.
-
-## How to use
-The process to set up and launch open_agb_firm is similar that of [GodMode9](https://github.com/d0k3/GodMode9).
-* Download the [latest release](https://github.com/profi200/open_agb_firm/releases/latest) and extract it to obtain `open_agb_firm.firm`.
-* Copy the `open_agb_firm.firm` file to your 3DS's SD card at `/luma/payloads` if using Luma3DS or elsewhere if using fastboot3DS.
-* Launch open_agb_firm using Luma3DS by holding START while booting your 3DS or assign the firm to a slot if you're using fastboot3DS.
-* After open_agb_firm launches, use the file browser to navigate to a .GBA ROM to run.
+## Setup
+* Download the [latest release](https://github.com/profi200/open_agb_firm/releases/latest) and extract it.
+* Copy the `open_agb_firm.firm` file to your 3DS's SD card at `/luma/payloads` if you're using Luma3DS or `/gm9/payloads` if you're using fastboot3DS.
+* Copy the `3ds` folder to the root of your 3DS's SD card. Merge folders if asked.
+* Launch open_agb_firm using Luma3DS by holding START while booting your 3DS or assign it to a slot if you're using fastboot3DS.
+* After open_agb_firm launches, use the file browser to navigate to a `.GBA` ROM to run.
 
 ## Controls
 A/B/L/R/START/SELECT - GBA buttons, respectively
 
-SELECT+Y - Dump screen output to `/3ds/open_agb_firm/texture_dump.bmp`*\
-*If the screen output freezes, press HOME to fix it. This is a hard to track down bug that will be fixed.
+SELECT+Y - Dump screen output to `/3ds/open_agb_firm/texture_dump.bmp`
+* If the screen output freezes, press HOME to fix it. This is a hard to track down bug that will be fixed.
 
 Hold the power button to turn off the 3DS.
 
 ## Configuration
-
 Settings are stored in `/3ds/open_agb_firm/config.ini`.
 
 ### General
-`u8 backlight` - Backlight brightness (default: `40`, max: `64`)
+General settings.
 
-`bool biosIntro` - Show GBA BIOS intro at game startup (default: `true`)
+`u8 backlight` - Backlight brightness
+* Default: `64`
+* Possible values:
+  * Old 3DS: `20`-`117`
+  * New 3DS: `16`-`142`
+
+`bool biosIntro` - Show GBA BIOS intro at game startup
+* Default: `true`
+
+`bool useGbaDb` - Use `gba_db.bin` to get save types
+* Default: `true`
 
 ### Video
-`float inGamma` - Set gamma (default: `2.2`)
+Video-related settings.
+
+`bool adjustGamma` - Adjust screen gamma using values from `inGamma`, `outGamma`, `contrast`, and `brightness`
```

Comment:
Is the default gamma setting equal to setting inGamma and outGamma to the same value? Even then, the `adjustGammaTableForGba()` function doesn't seem to use `REG_LCD_PDC0_GTBL_FIFO` at all as input, while the `adjustGamma` setting leaves it preserved.

Reply:
You are misunderstanding how it works. The default slightly darkens the screen to account for the bad calibration. If inGamma and outGamma are the same value it does nothing.

"Even then, the adjustGammaTableForGba() function doesn't seem to use REG_LCD_PDC0_GTBL_FIFO at all as input, while setting the adjustGamma setting to false leaves its initial value preserved." I have no idea what you are talking about.

Reply:
Of course it is. It's basically just 3 linearly incrementing bytes (identity mapping): https://github.com/profi200/open_agb_firm/blob/kernel_experiments/source/arm11/hardware/gfx.c#L358

When inGamma and outGamma are the same it's basically a no op and you get the same incrementing values. What adjustGammaTableForGba() does is gamma en-/decoding + brightness and contrast.

I made a table with graph a while ago that shows what the values do. Stupid Microsoft limited uploads to their own, proprietary formats so i had to zip it.
[GBA_colors.zip](https://github.com/profi200/open_agb_firm/files/6116722/GBA_colors.zip)

Reply:
I see. Still seems like a messy workaround to get the original LCD colors, though, as the adjustGammaTable function would still be called and the for loop would still be iterated through.

Reply:
For actual real GBA LCD emulation i would need to apply a filter like this:
https://github.com/mgba-emu/mgba/blob/master/res/shaders/gba-color.shader/gba-color.fs

Problem is there are no fragment shaders on the GPU so i have to resort to very inaccurate texenv (integer math).

Reply:
Okay. Can I leave the adjustGamma option though? It's still a useful way to get around the gamma adjustment during the loading process and preserve gamma settings in the `config.ini` file in case the user wants to use them again.